### ISSUE TITLE
Add a finetuning notebook to the inference repo

### ISSUE THREAD

**Issue by JLrumberger** [angelolab/Nimbus-Inference#27]
**Relevant background**

Add functionality to the Nimbus and Dataset class to allow for easy fine-tuning, performance evaluation and 

**Design overview**

Give a high-level description of what the proposed project will actually do

**Code mockup**

Provide a more detailed description of how the project will implement the above functionality. This should describe the logical flow of the program, detailing how the different parts interact with each other. It should list the specific helper functions that will be created, either with a description of what they'll do, or with psuedo-code for how they'll be implemented. 

**Required inputs**

Provided a description of the required inputs for this project, including column names for dfs, dimensions for image data, prompts for user input, directory structure for loading data, etc

**Output files**

Provide a description of the outputs for this project. If any plots will be generated, provide (simple) sketches demonstrating the plot type and axes labels. 

**Timeline**
Give a rough estimate for how long you think the project will take. In general, it's better to be too conservative rather than too optimistic. 
- [ ] A couple days
- [ ] A week
- [ ] Multiple weeks. For large projects, make sure to agree on a plan that isn't just a single monster PR at the end. 

Estimated date when a fully implemented version will be ready for review:

Estimated date when the finalized project will be merged in:
